### ISSUE TITLE
Hash-pinned GitHub Actions aren't updated if they're tag-pinned elsewhere

### ISSUE THREAD

**Issue by pnacht** [dependabot/dependabot-core#7419]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

github-actions

### Manifest location and content before the Dependabot update

All of the hash-pinned Actions in https://github.com/pnacht/harfbuzz/blob/main/.github/workflows/linux-ci.yml (MWE'd below) are out-of-date:

```yml
name: linux-ci

on: 
  push:

permissions:
  contents: read

jobs:
  build:
    runs-on: ubuntu-20.04

    steps:
    - name: Checkout
      uses: actions/checkout@83b7061638ee4956cf7545a6f7efe594e5ad0247 # v3.5.1
    - name: Setup Ccache
      uses: hendrikmuhs/ccache-action@fba817f3c0db4f854d7a3ee688241d6754da166e # v1.2.8
    - name: Upload Coverage
      uses: codecov/codecov-action@894ff025c7b54547a9a2a1e9f228beae737ad3c2 # v3.1.3
```

### dependabot.yml content

https://github.com/pnacht/harfbuzz/blob/main/.github/dependabot.yml:

```yml
version: 2
updates:
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "weekly"
  - package-ecosystem: "pip"
    directory: "/.ci"
    schedule:
      interval: "weekly"
```

### Updated dependency

None of the Actions were updated

### What you expected to see, versus what you actually saw

I expected to receive PRs updating each of the Actions to their most up-to-date version, such that the end result would be:

```yml
    steps:
    - name: Checkout
      uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
    - name: Setup Ccache
      uses: hendrikmuhs/ccache-action@ca3acd2731eef11f1572ccb126356c2f9298d35e # v1.2.9
    - name: Upload Coverage
      uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d # v3.1.4
```

### Images of the diff or a link to the PR, issue, or logs

Logs:

```
  proxy | 2023/06/09 13:57:17 proxy starting, commit: e11a9091e6d61fc45afc46a500c4b9a417032297
  proxy | 2023/06/09 13:57:17 Listening (:1080)
updater | 2023-06-09T13:57:17.535605098 [676022561:main:WARN:src/devices/src/legacy/serial.rs:222] Detached the serial input due to peer close/error.
updater | time="2023-06-09T13:57:19Z" level=info msg="guest starting" commit=46321cdb8407cc5cd7b86abb8137007376af9f3a
updater | time="2023-06-09T13:57:19Z" level=info msg="starting job..." fetcher_timeout=10m0s job_id=676022561 updater_timeout=45m0s updater_version=d89ea263e203b51f96c6814ebd1a93a69759d7a8-github-actions
updater | 2023/06/09 13:57:21 INFO Raven 3.1.2 ready to catch errors
updater | 2023/06/09 13:57:21 INFO <job_676022561> Starting job processing
  proxy | 2023/06/09 13:57:21 [002] GET https://api.github.com:443/repos/pnacht/harfbuzz
  proxy | 2023/06/09 13:57:21 [002] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [002] 200 https://api.github.com:443/repos/pnacht/harfbuzz
  proxy | 2023/06/09 13:57:22 [004] GET https://api.github.com:443/repos/pnacht/harfbuzz/git/refs/heads/main
  proxy | 2023/06/09 13:57:22 [004] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [004] 200 https://api.github.com:443/repos/pnacht/harfbuzz/git/refs/heads/main
  proxy | 2023/06/09 13:57:22 [006] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [006] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [006] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [008] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [008] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [008] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [010] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/arm-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [010] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [010] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/arm-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [012] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/cifuzz.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [012] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [012] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/cifuzz.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [014] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/configs-build.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [014] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [014] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/configs-build.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [016] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/coverity-scan.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [016] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [016] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/coverity-scan.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [018] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/linux-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [018] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:22 [018] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/linux-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [020] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/macos-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:22 [020] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:23 [020] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/macos-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:23 [022] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/msvc-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:23 [022] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:23 [022] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/msvc-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:23 [024] GET https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/msys2-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
  proxy | 2023/06/09 13:57:23 [024] * authenticating github api request with token for api.github.com
  proxy | 2023/06/09 13:57:23 [024] 200 https://api.github.com:443/repos/pnacht/harfbuzz/contents/.github/workflows/msys2-ci.yml?ref=875455b2e9e178154e4d6fd2a3f1b06737045b1b
updater | 2023/06/09 13:57:23 INFO <job_676022561> Finished job processing
updater | time="2023-06-09T13:57:23Z" level=info msg="task complete" container_id=job-676022561-file-fetcher exit_code=0 job_id=676022561 step=fetcher
updater | 2023/06/09 13:57:24 INFO Raven 3.1.2 ready to catch errors
updater | 2023/06/09 13:57:24 INFO <job_676022561> Starting job processing
  proxy | 2023/06/09 13:57:25 [026] GET https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [026] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:25 [026] 200 https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [028] GET https://github.com:443/google/oss-fuzz.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [028] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:25 [028] 200 https://github.com:443/google/oss-fuzz.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [030] GET https://github.com:443/google/oss-fuzz.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [030] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:25 [030] 200 https://github.com:443/google/oss-fuzz.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [032] GET https://github.com:443/actions/upload-artifact.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [032] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:25 [032] 200 https://github.com:443/actions/upload-artifact.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [034] GET https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [034] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:25 [034] 200 https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [036] GET https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:25 [036] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:26 [036] 200 https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [038] GET https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [038] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:26 [038] 200 https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [040] GET https://github.com:443/hendrikmuhs/ccache-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [040] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:26 [040] 200 https://github.com:443/hendrikmuhs/ccache-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [042] GET https://github.com:443/codecov/codecov-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [042] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:26 [042] 200 https://github.com:443/codecov/codecov-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [044] GET https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [044] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:26 [044] 200 https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [046] GET https://github.com:443/hendrikmuhs/ccache-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [046] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:26 [046] 200 https://github.com:443/hendrikmuhs/ccache-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [048] GET https://github.com:443/codecov/codecov-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [048] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:26 [048] 200 https://github.com:443/codecov/codecov-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [050] GET https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:26 [050] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:27 [050] 200 https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [052] GET https://github.com:443/hendrikmuhs/ccache-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [052] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:27 [052] 200 https://github.com:443/hendrikmuhs/ccache-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [054] GET https://github.com:443/actions/setup-python.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [054] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:27 [054] 200 https://github.com:443/actions/setup-python.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [056] GET https://github.com:443/ilammy/msvc-dev-cmd.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [056] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:27 [056] 200 https://github.com:443/ilammy/msvc-dev-cmd.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [058] GET https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [058] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:27 [058] 200 https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [060] GET https://github.com:443/msys2/setup-msys2.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:27 [060] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:27 [060] 200 https://github.com:443/msys2/setup-msys2.git/info/refs?service=git-upload-pack
updater | 2023/06/09 13:57:28 INFO <job_676022561> Starting update job for pnacht/harfbuzz
updater | 2023/06/09 13:57:28 INFO <job_676022561> Checking all dependencies for version updates...
updater | 2023/06/09 13:57:28 INFO <job_676022561> Checking if actions/upload-artifact 3 needs updating
  proxy | 2023/06/09 13:57:28 [066] GET https://github.com:443/actions/upload-artifact.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:28 [066] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:28 [066] 200 https://github.com:443/actions/upload-artifact.git/info/refs?service=git-upload-pack
updater | 2023/06/09 13:57:28 INFO <job_676022561> Latest version is 3
updater | 2023/06/09 13:57:28 INFO <job_676022561> No update needed for actions/upload-artifact 3
updater | 2023/06/09 13:57:28 INFO <job_676022561> Checking if actions/checkout 3 needs updating
  proxy | 2023/06/09 13:57:28 [068] GET https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:28 [068] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:28 [068] 200 https://github.com:443/actions/checkout.git/info/refs?service=git-upload-pack
updater | 2023/06/09 13:57:28 INFO <job_676022561> Latest version is 3
updater | 2023/06/09 13:57:28 INFO <job_676022561> No update needed for actions/checkout 3
updater | 2023/06/09 13:57:28 INFO <job_676022561> Checking if ilammy/msvc-dev-cmd 1 needs updating
  proxy | 2023/06/09 13:57:28 [070] GET https://github.com:443/ilammy/msvc-dev-cmd.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:28 [070] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:28 [070] 200 https://github.com:443/ilammy/msvc-dev-cmd.git/info/refs?service=git-upload-pack
updater | 2023/06/09 13:57:28 INFO <job_676022561> Latest version is 1
updater | 2023/06/09 13:57:28 INFO <job_676022561> No update needed for ilammy/msvc-dev-cmd 1
updater | 2023/06/09 13:57:28 INFO <job_676022561> Checking if hendrikmuhs/ccache-action 1.2 needs updating
  proxy | 2023/06/09 13:57:28 [072] GET https://github.com:443/hendrikmuhs/ccache-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:28 [072] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:28 [072] 200 https://github.com:443/hendrikmuhs/ccache-action.git/info/refs?service=git-upload-pack
updater | 2023/06/09 13:57:28 INFO <job_676022561> Latest version is 1.2
updater | 2023/06/09 13:57:28 INFO <job_676022561> No update needed for hendrikmuhs/ccache-action 1.2
updater | 2023/06/09 13:57:28 INFO <job_676022561> Checking if msys2/setup-msys2 2 needs updating
  proxy | 2023/06/09 13:57:28 [074] GET https://github.com:443/msys2/setup-msys2.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:28 [074] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:28 [074] 200 https://github.com:443/msys2/setup-msys2.git/info/refs?service=git-upload-pack
updater | 2023/06/09 13:57:28 INFO <job_676022561> Latest version is 2
updater | 2023/06/09 13:57:28 INFO <job_676022561> No update needed for msys2/setup-msys2 2
updater | 2023/06/09 13:57:28 INFO <job_676022561> Checking if actions/setup-python 4 needs updating
  proxy | 2023/06/09 13:57:28 [076] GET https://github.com:443/actions/setup-python.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:28 [076] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:29 [076] 200 https://github.com:443/actions/setup-python.git/info/refs?service=git-upload-pack
updater | 2023/06/09 13:57:29 INFO <job_676022561> Latest version is 4
updater | 2023/06/09 13:57:29 INFO <job_676022561> No update needed for actions/setup-python 4
updater | 2023/06/09 13:57:29 INFO <job_676022561> Checking if codecov/codecov-action 3 needs updating
  proxy | 2023/06/09 13:57:29 [078] GET https://github.com:443/codecov/codecov-action.git/info/refs?service=git-upload-pack
  proxy | 2023/06/09 13:57:29 [078] * authenticating git server request (host: github.com)
  proxy | 2023/06/09 13:57:29 [078] 200 https://github.com:443/codecov/codecov-action.git/info/refs?service=git-upload-pack
updater | 2023/06/09 13:57:29 INFO <job_676022561> Latest version is 3
updater | 2023/06/09 13:57:29 INFO <job_676022561> No update needed for codecov/codecov-action 3
updater | 2023/06/09 13:57:29 INFO <job_676022561> Finished job processing
updater | time="2023-06-09T13:57:29Z" level=info msg="task complete" container_id=job-676022561-updater exit_code=0 job_id=676022561 step=updater
```

### Smallest manifest that reproduces the issue

I haven't tested this, but I assume this would demonstrate the issue:

```yml
# .github/workflows/tagged.yml
on: push

jobs:
  job:
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@v3
```

```yml
# .github/workflows/hashed.yml
on: push

jobs:
  job:
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@83b7061638ee4956cf7545a6f7efe594e5ad0247 # v3.5.1
```

**Comment by pnacht** [comment]
I don't know if this is related to https://github.com/dependabot/dependabot-core/issues/4781#issuecomment-1058064268. There are other workflows that haven't been hash-pinned.

Ironically, I'm trying to show the original harfbuzz repository that dependabot can help update hash-pinned GitHub Actions and so just changed one file as an example. I'll try to change all the other workflows too...

**Comment by deivid-rodriguez** [comment]
This was fixed by #8068 and #8098 :tada:
